Tests/Common/Util/StreamReaderExtensionsTests.cs

```diff
@@ -113,6 +113,31 @@ public void GetDecimalEmptyString()
             Assert.AreEqual(0, smartStream.GetDecimal());
         }
 
+        [Test]
+        public void GetNextCsv()
+        {
+            var stream = "1,2,3,\"hello\",\"world\"\r\n3,2,1,\"goodbye\",\"world\"\n2,3,1,\"hello again, world!\",\"goodbye again, world!\"\r\n".ToStream();
```

Comment:
Thanks for the unit test!
I think we need a few more though, it will fail if the stream does not end in `\r\n`, please test with different possible endings `,` or `"`, etc

Reply:
*Think its missing to check the `NoMoreData` case